src/pipecat/services/speechmatics/stt.py

```diff
@@ -0,0 +1,787 @@
+        __version__,
+    )
+except ModuleNotFoundError as e:
+    logger.error(f"Exception: {e}")
+    logger.error(
+        "In order to use Speechmatics, you need to `pip install pipecat-ai[speechmatics]`."
+    )
+    raise Exception(f"Missing module: {e}")
+
+
+class AudioBuffer:
+    """Audio buffer for STT clients.
+
+    The Python SDK expects audio in a pre-defined number of frames. This
+    buffer will accumulate the data from the pipeline and provide it to the
+    STT client in the correct lengths, waiting for the number of frames to
+    be available.
+
+    Args:
+        maxsize: Maximum size of the buffer.
+    """
+
+    def __init__(self, maxsize: int = 0):
+        self._queue = asyncio.Queue(maxsize=maxsize)
+        self._current_chunk = b""
+        self._position = 0
+        self._closed = False
+
+    def write_audio(self, data: bytes) -> None:
+        """Write audio data to the buffer (thread-safe).
+
+        Args:
+            data: Audio data to write.
+        """
+        if data:
+            try:
+                self._queue.put_nowait(data)
+            except asyncio.QueueFull:
+                pass
+
+    async def read(self, size: int) -> bytes:
+        """Read exactly `size` bytes from the buffer (thread-safe).
+
+        This process will block until the required number of bytes are available
+        in the buffer. Audio is received from the pipeline in varying sizes, so
+        this buffer will accumulate the data and provide it to the STT client in
+        the correct lengths, waiting for the number of frames to be available.
+
+        Calling stop() will close the buffer and release the blocking read
+        process.
+
+        Args:
+            size: Number of bytes to read.
+
+        Returns:
+            bytes: Audio data read from the buffer.
+        """
+        result = b""
+        bytes_needed = size
+
+        while bytes_needed > 0 and not self._closed:
+            # Use data from current chunk if available
+            if self._position < len(self._current_chunk):
+                available = len(self._current_chunk) - self._position
+                take = min(bytes_needed, available)
+                result += self._current_chunk[self._position : self._position + take]
+                self._position += take
+                bytes_needed -= take
+                continue
+
+            # Get next chunk
+            try:
+                chunk = await asyncio.wait_for(self._queue.get(), timeout=0.1)
+                if chunk is None:
+                    continue
+                self._current_chunk = chunk
+                self._position = 0
+            except asyncio.TimeoutError:
+                await asyncio.sleep(0)
+                continue
+
+        return result
+
+    def stop(self) -> None:
+        """Close the audio buffer."""
+        self._closed = True
+
+
+@dataclass
+class SpeechFragment:
+    """Fragment of an utterance.
+
+    Attributes:
```

Comment:
dataclasses should list members as `Parameters`, which is mainly to do with how sphinx and autodoc generate HTML. It's a major rough edge of working with the docs autogeneration.

```suggestion
    Parameters:
```